pkg/kubelet/volumemanager/reconciler/reconciler.go

```diff
@@ -426,7 +402,34 @@ func (rc *reconciler) syncStates(podsDir string) {
 
 }
 
-// Reconstruct Volume object and reconstructedVolume data structure by reading the pod's volume directories
+func (rc *reconciler) cleanupMounts(volume podVolume) {
+	if rc.desiredStateOfWorld.VolumeExistsWithSpecName(volume.podName, volume.volumeSpecName) {
+		return
+	}
+	glog.V(2).Infof("Reconciler sync states: could not find information (PID: %s) (Volume SpecName: %s) in desired state, clean up the mount points",
+		volume.podName, volume.volumeSpecName)
+	mountedVolume := operationexecutor.MountedVolume{
+		PodName:             volume.podName,
+		VolumeName:          v1.UniqueVolumeName(volume.volumeSpecName),
+		InnerVolumeSpecName: volume.volumeSpecName,
+		PluginName:          volume.pluginName,
+		PodUID:              types.UID(volume.podName),
+	}
+	volumeHandler, err := operationexecutor.NewVolumeHandlerWithMode(volume.volumeMode, rc.operationExecutor)
+	if err != nil {
+		glog.Errorf(mountedVolume.GenerateErrorDetailed(fmt.Sprintf("operationExecutor.NewVolumeHandler for UnmountVolume failed"), err).Error())
+		return
+	}
+	// TODO: Currently cleanupMounts only includes UnmountVolume operation. In the next PR, we will add
+	// to unmount both volume and device in the same routine.
+	err = volumeHandler.UnmountVolumeHandler(mountedVolume, rc.actualStateOfWorld)
```

Comment:
@jingxu97 

GenerateUnmapVolumeFunc() depends on volumeSpec to get global map path using GetGlobalMapPath()
Do you plan to change implementation of GetGlobalMapPath() in operation_generator and plugins too?
Otherwise, unmap volume hit panic with null volume spec reference in plugin.

Reply:
Thanks for pointing this out. I added the globalMountPath in the MountedVolume data structure so it can directly get the path instead of constructing from VolumeSpec.

Reply:
I see. In that case, we don't need to change plugins.